core_impl/src/main/java/io/opencensus/stats/MutableAggregation.java

```diff
@@ -0,0 +1,348 @@
+/*
+ * Copyright 2017, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.stats;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import io.opencensus.common.Function;
+
+/** Mutable version of {@link Aggregation} that supports adding values. */
+abstract class MutableAggregation {
+
+  private MutableAggregation() {
+  }
+
+  /**
+   * Put a new value into the MutableAggregation.
+   *
+   * @param value new value to be added to population
+   */
+  abstract void add(double value);
+
+  /**
+   * Applies the given match function to the underlying data type.
+   */
+  abstract <T> T match(
+      Function<? super MutableSum, T> p0,
+      Function<? super MutableCount, T> p1,
+      Function<? super MutableHistogram, T> p2,
+      Function<? super MutableRange, T> p3,
+      Function<? super MutableMean, T> p4,
+      Function<? super MutableStdDev, T> p5,
+      Function<? super MutableAggregation, T> defaultFunction);
```

Comment:
I don't think we need a default case, because this method is internal, and it can be updated when the subclasses are updated.